### DISCUSSION TITLE
Auto Completion with DNF Packages

### DISCUSSION THREAD

**Discussion by Tahinli** [nushell/nushell#12743] [Q&A] [answered]
Hi, is there a way to enable autocompletion for dnf packages like in the bash:
![image](https://github.com/nushell/nushell/assets/96421894/553f1d73-d82e-4663-95c8-78b05cb4ec9c)

**Comment by ghost** [comment, marked as answer]
in nushell you have to write completions for external programs yourself (or copy someone elses solution).  
it seems as if noone has published a usable one for dnf yet ([only this](https://github.com/nushell/nu_scripts/blob/main/custom-completions/auto-generate/completions/dnf.nu)).

i have never used dnf, but to give you an idea how it could roughly look in your configuration heres is a guess:

```nu
def "complete dnf pacakges" [] {
  dnf list --all | lines
}

export extern "dnf install" [
  --yes(-y)  # accept all prompts
  ...packages: string@"complete dnf packages"
]
```

**Reply by Tahinli** [reply]
thanks

**Comment by devyn** [comment]
You can also look at external completers. I think the most popular one is `carapace`. If you set it up then you can get completions for anything it supports.
